### PR TITLE
Fix a bug in sample code.

### PR DESCRIPTION
**Pull request by lifuhuang**
The original code sample was wrong and would throw run-time error. That's because `keeper`, as an instance property, is not in the prototype of the instance's constructor, so I changed `keeper` to be a getter of `Animal`/`Bee`.

Fixes #

### CODE REVIEW

**Comment by mhegazy** [comment]
The `findKeeper` seems like a convoluted way to show generics. i would change the function instead to take an instance instead of the class. i.e. `findKeeper(new Lion())`

**Comment by lifuhuang** [comment]
Sorry for the late response. A new commit has been made.

File: pages/Generics.md [hunk 1]
```diff
@@ -309,18 +309,22 @@ class Animal {
...
+    get keeper() {
+        return this._keeper;
+    }
 }
 
-function findKeeper<A extends Animal, K> (a: {new(): A;
-    prototype: {keeper: K}}): K {
-
-    return a.prototype.keeper;
+function findKeeper<A extends Animal, K> (a: A & { keeper: K }): K {
```
Comment (mhegazy) [inline pages/Generics.md]: not sure why you need the animal all togather.. 

```ts
function findKeeper<T> (a: { keeper: T }): T
```

File: pages/Generics.md [hunk 2]
```diff
@@ -309,18 +309,22 @@ class Animal {
...
-    keeper: BeeKeeper;
+    private _keeper = new BeeKeeper();
+    get keeper() {
+        return this._keeper;
+    };
 }
 
 class Lion extends Animal {
-    keeper: ZooKeeper;
+    private _keeper = new ZooKeeper();
```
Comment (mhegazy) [inline pages/Generics.md]: no need for complicating the class by adding a private fields and a getter, just keep the property declaration that was there before: ` keeper: ZooKeeper`.

File: pages/Generics.md [hunk 3]
```diff
@@ -309,18 +309,22 @@ class Animal {
 }
 
 class Bee extends Animal {
-    keeper: BeeKeeper;
+    private _keeper = new BeeKeeper();
```
Comment (mhegazy) [inline pages/Generics.md]: no need for complicating the class by adding a private fields and a getter, just keep the property declaration that was there before: ` keeper: BeeKeeper`.

**Comment by mhegazy** [comment]
hmm.. looks like something went wrong there.. github is not happy :(

image

**Pull request closed without merging**